modules/standard/Regexp.chpl

```diff
@@ -450,12 +450,9 @@ class BadRegexpError : Error {
                    this flag can be set inside the regular expression with
                    ``(?U)``.
 
+   :throws: BadRegexError: If CHPL_REGEXP is set to none rather than re2.
```

Comment:
Syntactical feedback: it should be `:throws BadRegexError:`, not `:throws: BadRegexError:`, you have an extra semi colon.

Substantive feedback: again, the compiler error does not throw, it is just a compilation error.  `CHPL_REGEXP` being set to none is what triggers the *compiler* error, not what triggers the `throw` later in the body (on line 492).  The condition that causes `BadRegexpError` to be thrown is the if statement on line 483.  To determine what causes *that* function to return false, you'll need to look at where it is defined (my guess is somewhere in the runtime).

Think of a compiler error as equivalent in severity to failing to include a semi colon or making a typo when trying to call a function - when something like that happens, the compiler doesn't have the tools to adjust for it and needs the person writing the program to make a change and try to compile again.  In this case, having CHPL_REGEXP set to none means the compiler isn't able to include the underlying implementation that supports the Regexp module at all.  This is in contrast to error handling constructs, where something has gone wrong but the user could potentially have written their program to detect the error and adjust as a result.

If you're still confused about error handling, I recommend checking out the [error handling primer](https://chapel-lang.org/docs/latest/primers/errorHandling.html).  You can also read the [chapter in the spec](https://chapel-lang.org/docs/latest/language/spec/error-handling.html).

Reply:
Hi @lydia-duncan Mam
I have read the [qio_regexp.h](https://github.com/chapel-lang/chapel/blob/master/runtime/include/qio/qio_regexp.h) file in `runtime` folder. Kindly refer to line [59](https://github.com/chapel-lang/chapel/blob/master/runtime/include/qio/qio_regexp.h#L59) of this file. Here it is mentioned that the function will `Return true for ok` and if false will return an error string. However, I am still not able to find the definition of this function and when will it return false.
So how should the comments look? Can you please help me with this?

Reply:
I think the function definition you're looking for is [here](https://github.com/chapel-lang/chapel/blob/master/runtime/src/qio/regexp/re2/re2-interface.cc#L203)

I found that file by looking in the `runtime/src` directory.  Since the .h file was under `qio` in the `include` directory, I looked in the `runtime/src/qio` directory, saw a directory that was labelled the same as the module we're looking at, and then from there I made a guess based on knowing that our Regexp implementation is supported by the package `re2` under the covers (which is definitely a leap that would be hard for a new person, sorry).   But I suspect a `grep` command in the runtime directory would have gotten to it faster.

You may have to do some more chasing - I think the main function that function relies upon is still in the same file, but it may be you'll have to look at something `re2` defines to fully determine the conditions.

Reply:
I have read the modules for `re2` and `regexp`, studied the function definition of the function we are working with. 
An error is thrown if `re2 && re2->ok()` evaluates to `false`. In [this](https://github.com/chapel-lang/chapel/blob/master/third-party/re2/re2-src/re2/re2.cc) file, in line [1022](https://github.com/chapel-lang/chapel/blob/master/third-party/re2/re2-src/re2/re2.cc#L1022), the error statement is `Invalid RE2`. So should this be the reason for throwing an error?
In my opinion, this error is related to the improper creation of `re2`, as mentioned [here](https://github.com/chapel-lang/chapel/blob/master/third-party/re2/re2-src/re2/re2.h#L286)

Reply:
I think that's a reasonable interpretation of the code, yeah!